x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/SecurityContext.java

```diff
@@ -86,6 +86,24 @@ public AuthorizationEngine.AuthorizationInfo getAuthorizationInfoFromContext() {
         return Objects.requireNonNull(threadContext.getTransient(AUTHORIZATION_INFO_KEY), "authorization info is missing from context");
     }
 
+    @Nullable
+    public AuthorizationEngine.ParentActionAuthorization getParentAuthorization() {
+        try {
+            return AuthorizationEngine.ParentActionAuthorization.readFromThreadContext(threadContext);
+        } catch (IOException e) {
+            logger.error("failed to read parent authorization from thread context", e);
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public void setParentAuthorization(AuthorizationEngine.ParentActionAuthorization parentAuthorization) {
+        try {
+            parentAuthorization.writeToThreadContext(threadContext);
+        } catch (IOException e) {
+            throw new AssertionError("failed to write parent authorization to the thread context", e);
```

Comment:
nit: maybe stick with the `UnckedIOException(e)` pattern, unless I'm missing the reasoning why this should be an assertion rather than a hard exception.